cmd/appengine/device.go

```diff
@@ -1634,32 +1635,42 @@ func parseSendDataPayload(payload string, mappingType interfaces.AstarteMappingT
 	case interfaces.BinaryBlobArray, interfaces.BooleanArray, interfaces.DateTimeArray, interfaces.DoubleArray,
 		interfaces.IntegerArray, interfaces.LongIntegerArray, interfaces.StringArray:
 
-		//wait it's all string?
-		payload = strings.Replace(payload, "[", "", -1)
-		payload = strings.Replace(payload, "]", "", -1)
-		payload_parsed := strings.Split(payload, ",")
-		//always has been
+		//check if payload is an empty array, bypass conversions and just return an empty array
+		// if it is not, proceed as usual
+		if payload == "[]" {
+			ret = make([]interface{}, 0)
 
-		jsonOut := make([]interface{}, len(payload_parsed))
-		for i, v := range payload_parsed {
-			jsonOut[i] = v
-		}
+		} else {
 
-		retArray := []interface{}{}
-		// Do a smarter conversion here.
-		for _, v := range jsonOut {
-			switch val := v.(type) {
-			case string:
-				p, err := parseSendDataPayload(strings.TrimSpace(val), interfaces.AstarteMappingType(strings.TrimSuffix(string(mappingType), "array")))
-				if err != nil {
-					return nil, err
+			//wait it's all string?
+			payload = strings.Replace(payload, "[", "", -1)
+			payload = strings.Replace(payload, "]", "", -1)
+			payload_parsed := strings.Split(payload, ",")
+			//always has been
```

Comment:
I would suggest using a different approach for parsing the array data. We know that:
- The input payload has the form `[<comma separated data>]`
- A `[<comma separated data>]` value is a valid JSON object
- Go's `json.Unmarshal` is able to handle these objects (see https://go.dev/play/p/ixaicx2S7cj).

Therefore, my suggestion is to let Go handle characters such as `[` and just do the conversion of payloads, something like:
```go
var jsonPayload []interface{}
if err := json.Unmarshal([]byte(payload), &jsonPayload); err != nil {
	fmt.Println(err) //Maybe also warn the user of the bad input payload
	System.exit(1)
}
retArray := []interface{}{}

// Do a smarter conversion here.
for _, v := range jsonPayload {
// etc etc, the rest is the same
```

This would mean that the user will have to write a valid JSON payload, e.g. strings inside arrays should be put in double quotes, like `astartectl appengine devices publish-datastream <device_id> <interface> <path> '["a", "b", "c"]'`

Reply:
after a irl discussion, the following points arose: 
- golang json.Unmarshal returns always a float64 interface, even with an array of integers, making the conversion a little bit challenging and type assertion non-trivial
- the above implementation is not the best and kinda ugly, but for the next release it gets the work done and the bug fixed
- a complete rewrite of parseSendDataPayload function is needed, in order to use the more stable golang conversion, but it's going to be another release and an issue will be opened